### ISSUE TITLE
[V3] Implement new Concept Exercise: sets

### ISSUE THREAD

**Issue by BethanyG** [exercism/python#2296]
This issue describes how to implement the `sets` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the basics of the [set](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/builtin_types/set.md) (set type) in Python.

## Learning objectives

*  understand that a set is an unordered collection of distinct hashable objects
*  create a `set` via constructor (`set()`) and  literal (`{}`)
*  de-dupe a list of elements by converting a sequence type such as a list to a set type
*  check for a membership of an element in a given set via `in`
*  set comparison functions and set comparison operators (`=<`, `>=`, `issubset()`, `issuperset()`, etc.)
*  additional set operators (`union`, `intersection`, `difference`, and `symmetric_difference`)
*  add values to a given set via `add()`
*  remove values from a given set via `discard()`
*  iterate through a given set by using `for item in set`

## Out of scope

*  `frozenset()`
*  `clear()` to delete all elements of a set
*  check the length of a given set via `len()`
*  `remove` as opposed to `discard` (_`remove` tosses a `keyerror` if the element is not present_)
*   all forms/variants of `update()`
*  remove (and use) a value from a given set via `pop()`
*  make shallow copy(s) of a given set via `copy()`
*  using additional builtins such as `sorted()`, `min()`, or `map()` with a set
*  set comprehensions

## Concepts

*  `sets`
*  [`hashable`](https://docs.python.org/3/glossary.html#term-hashable) objects
* `set` comparisons
* `set` operations

## Prerequisites

* `basics`
* `booleans`
*  `comparisons`
*  `dicts`
* `lists`
* `loops`

## Resources to refer to

*  [Set Types (Python Official Docs)](https://docs.python.org/3/library/stdtypes.html#set-types-set-frozenset)
*   [Hashable (Python Official Docs Glossary)](https://docs.python.org/3/glossary.html#term-hashable)
*  [immutable (Python Official Docs Glossary)](https://docs.python.org/3/glossary.html#term-immutable)  

### Hints

Hints should link to the `Sets` section of the Python docs tutorial: [Sets](https://docs.python.org/3/tutorial/datastructures.html#sets)

### After

After, the student can explore comprehension syntax, although it will be taught in separate exercises. This would also be a good time to explore set comparisons via function &/or operator, or experimenting with the `issuperset()` & `issubset()` functions.

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase, and the test file named dicts_basic_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
- edited for updated naming by @yawpitch
